arch/arm: cortex_m: Disable DWT null-pointer exception detection

erwango opened the pull request:
Null-pointer exception detection using DWT is currently incompatible
with current openocd runner default implementation that leaves debug
mode on by default.
As a consequence, on all targets that use openocd runner, null-pointer
exception detection using DWT will generated an assert.
As a consequence, all tests are failing on such platforms.

Disable this until openocd behavior is fixed (#32984)

Rationale: https://github.com/zephyrproject-rtos/zephyr/pull/31481 is a great addition, but given this is making a wide range of boards unusable, I think it's activation can wait https://github.com/zephyrproject-rtos/zephyr/issues/32984 to be fixed.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

EDIT: Following discussion with @ioannisg and @galak, enable DWT based solution on Nordic based platforms

erwango committed:
arch/arm: cortex_m: Disable DWT based null-pointer exception detection

Null-pointer exception detection using DWT is currently incompatible
with current openocd runner default implementation that leaves debug
mode on by default.
As a consequence, on all targets that use openocd runner, null-pointer
exception detection using DWT will generated an assert.
As a consequence, all tests are failing on such platforms.

Disable this until openocd behavior is fixed (#32984) and enable
the MPU based solution for now.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ioannisg reviewed (commented):
This is as good as it can get right now, @erwango thanks for the leaving the MPU-solution enabled; I guess this has nothing to do with openOCD.

erwango commented:
> This is as good as it can get right now, @erwango thanks for the leaving the MPU-solution enabled; I guess this has nothing to do with openOCD.

@ioannisg can you formally approve then?

erwango committed:
soc: nrf: Enable default DWT null pointer exception detection

Default null pointer exception detection to the DWT solution
instead of MPU based solution.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

nashif commented:
this fixes this issue for me https://github.com/zephyrproject-rtos/zephyr/issues/32966
